extendr-api/src/robj/from_robj.rs

```diff
@@ -289,3 +289,59 @@ where
         }
     }
 }
+
+#[cfg(feature = "faer")]
+impl<'a> FromRobj<'a> for faer::Mat<f64> {
+    fn from_robj(robj: &'a Robj) -> std::result::Result<Self, &'static str> {
+        use crate::scalar::Scalar;
+        if robj.is_matrix() {
+            if let Some(dim) = robj.dim() {
+                let dim: Vec<_> = dim.iter().map(|d| d.inner() as usize).collect();
+
+                if dim.len() != 2 {
+                    return Err("could not convert to matrix");
+                }
+
+                if let Some(slice) = robj.as_real_slice() {
+                    let fmat = faer::mat::from_column_major_slice::<f64>(&slice, dim[0], dim[1]);
+                    Ok(fmat.to_owned())
+                } else if let Some(slice) = robj.as_integer_slice() {
+                    let fmat = faer::Mat::<f64>::from_fn(dim[0], dim[1], |i, j| {
+                        slice[i + j * dim[0]] as f64
```

Comment:
Does this imply a copy?

Reply:
From my understanding of [faer::mat::from_column_major_slice](https://docs.rs/faer/0.17.1/faer/mat/fn.from_column_major_slice.html), it uses the same values. So the current flow uses that if the values are already f64 and if not, does copy with the cast to f64.

Reply:
Okay, would the .to_owned() just copy the reference to the structure? Sorry if this is a very basic question as rust is quite new to me.

Reply:
Ah, I see what you mean. I've asked and it does indeed copy, the alternative would be to return instead of `Mat` returning a `MatRef`. That however would not work for the integer branch since the returned value referencing `Mat` would be data owned by the current function. Maybe it would be worth it to also have a `impl<'a> FromRobj<'a> for MatRef<'a, f64>` that would only work for floats from R.

Reply:
That sounds like a great idea, being able to have a reference if the matrix is huge will be good both for performance and memory usage.

Reply:
I agree, addressed in commit https://github.com/extendr/extendr/pull/706/commits/6e14fe3078a4654e052fe1fd7a4a845aa48d44b5.